src/modm/processing/fiber/asm/arm_cm0.S

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+.file "arm_cm0.S"
+.text
+.globl modm_makecontext, modm_jumpcontext
+.align 2
+.type modm_makecontext,%function
+.type modm_jumpcontext,%function
+.syntax unified
+
+/* Implementation of modm_makecontext / modm_jumpcontext for Cortex-M0(+) based boards.
+ *
+ * Important note: We are not preserving r8-r11 because they are quite useless on Cortex-M0
+ * and storing them takes a lot of cycles (as they aren't supported by the push instruction).
```

Comment:
That sounds like bugs, bugs, bugs. ;-P Wait till you get to Cortex-M4 with FPU state, it's going to blow your ~mind~ context…

Reply:
Yeah, this was not a good idea.